packages/server/src/fhir/operations/utils/asyncjobexecutor.ts

```diff
@@ -0,0 +1,41 @@
+import { AsyncJob } from '@medplum/fhirtypes';
+import { Repository, systemRepo } from '../../repo';
+
+export class AsyncJobExecutor {
+  readonly repo: Repository;
+  private resource: AsyncJob | undefined;
+  constructor(repo: Repository) {
+    this.repo = repo;
+  }
+
+  async start(url: string): Promise<AsyncJob> {
```

Comment:
This doesn't actually "start" the job, so maybe call this `init`?

Then as previously mentioned, maybe add a separate `start` method which is not `async` and does not return a `Promise`.  It can also take the `callback` param, and call `run` with the standard `then` and `catch` handlers.